### PR TITLE
[PATCH v3] api: split packet, pktio, and random types into separate header files

### PR DESCRIPTION
**Pull request by MatiasElo**
Split API types into a separate header files to enable easier function
inlining. No actual API changes.

### CODE REVIEW

File: platform/linux-generic/include-abi/odp/api/abi/pool.h [hunk 1]
```diff
@@ -17,23 +18,7 @@
...
-typedef ODP_HANDLE_T(odp_pool_t);
-
-#define ODP_POOL_INVALID _odp_cast_scalar(odp_pool_t, 0)
-
-#define ODP_POOL_NAME_LEN  32
-
-/**
- * @}
- */
+/* Empty placeholder header for inline functions */
```
Comment (TuomasTaipale) [inline platform/linux-generic/include-abi/odp/api/abi/pool.h:21]: Nitpick: some empty header files comment `/* Empty header required due to the packet inline functions */` and some `/* Empty placeholder header for inline functions */`, for different purpose or just use the same in all?

File: include/odp/api/abi-default/pool_types.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+ */
+
+#ifndef ODP_ABI_POOL_TYPES_H_
+#define ODP_ABI_POOL_TYPES_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <stdint.h>
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply include/odp/api/abi-default/pool_types.h:15]: Ok, will be removed in V2.

File: platform/linux-generic/include-abi/odp/api/abi/pool_types.h [hunk 3]
```diff
@@ -0,0 +1,41 @@
...
+ */
+
+#ifndef ODP_API_ABI_POOL_TYPES_H_
+#define ODP_API_ABI_POOL_TYPES_H_
+
+#ifdef __cplusplus
+extern "C" {
+#endif
+
+#include <odp/api/std_types.h>
```
[the remark this answers is not in the document]
Comment (MatiasElo) [reply platform/linux-generic/include-abi/odp/api/abi/pool_types.h:20]: Will be removed in V2.

File: platform/linux-generic/include-abi/odp/api/abi/pool.h (hunk 1 above)
Comment (MatiasElo) [reply platform/linux-generic/include-abi/odp/api/abi/pool.h:21]: There is a small difference. In ABI compatible mode `pool.h` header is picked from `include/odp/api/abi-default/` and this header will remain empty also in the future. In non-ABI compatible mode `platform/linux-generic/include-abi/odp/api/abi/pool.h` is used instead. When we want to inline pool functions, the inline header will be included using this file (e.g. `platform/linux-generic/include-abi/odp/api/abi/queue.h`).

**Review by TuomasTaipale — APPROVED** [review]

**Review by psavol — APPROVED** [review]

**Pull request merged**